base_layer/wallet/src/wallet.rs

```diff
@@ -221,8 +220,8 @@ where
 
     /// This method consumes the wallet so that the handles are dropped which will result in the services async loops
     /// exiting.
-    pub async fn shutdown(mut self) {
-        self.shutdown.trigger().expect(
+    pub async fn shutdown(self, mut shutdown: Shutdown) {
```

Comment:
I propose that we rather change this into a wait_for_shutdown(...) and have the client's trigger the shutdown. So in the FFI lib you will need to do the following (In the Wallet FFI case we do not want to panic if the shutdown triggers are wrong so not using an expect() there)

```Rust
pub unsafe extern "C" fn wallet_destroy(wallet: *mut TariWallet) {
    if !wallet.is_null() {
        let mut m = Box::from_raw(wallet);
        match m.shutdown.trigger() {
            Err(_) => error!(target: LOG_TARGET, "No one is listening for the shutdown trigger!"),
            Ok(()) =>  m.runtime.block_on(m.wallet.wait_for_shutdown()),
            
        }
    }
}
```
And the equivalent in the console wallet and also the base node while there is still a wallet in there.

Reply:
Will do